cmd/soroban-cli/src/commands/network/start.rs

```diff
@@ -0,0 +1,196 @@
+use std::collections::HashMap;
+
+use bollard::{
+    container::{Config, CreateContainerOptions, StartContainerOptions},
+    image::CreateImageOptions,
+    service::{HostConfig, PortBinding},
+};
+use futures_util::TryStreamExt;
+
+use crate::commands::network::shared::{connect_to_docker, Network, DOCKER_SOCKET_PATH_HELP};
+
+const DEFAULT_PORT_MAPPING: &str = "8000:8000";
+const DOCKER_IMAGE: &str = "docker.io/stellar/quickstart";
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("⛔ ️Failed to start container: {0}")]
+    StartContainerError(#[from] bollard::errors::Error),
+}
+
+#[derive(Debug, clap::Parser, Clone)]
+pub struct Cmd {
+    /// Network to start
+    pub network: Network,
+
+    #[arg(short = 'd', long, help = DOCKER_SOCKET_PATH_HELP)]
+    pub docker_socket_path: Option<String>,
```

Comment:
I just changed this to `docker_host`, but am now noticing a small issue/nuance with how bollard is understanding the `DOCKER_HOST` env var in its current release. It seems to only consider `DOCKER_HOST` for HTTP, https and tcp connections, not socket connections. 

That's why I think I made this `docker_socket_path` originally, so that it was clear this was only to be used for socket connections. I think this may be confusing for users, since they may assume that they can use the `DOCKER_HOST` var like you can for the CLI, which allows it to be for socket, http, tcp, etc connections.

Bollard made some [updates](https://github.com/fussybeaver/bollard/commit/ba8a0ec415a878405fd3c90bad6dd2821190709c) in `master` that haven't been released yet and included a func called `connect_with_defaults`, but this still ignores the case that I was trying to account for... if the user has a non-standard socket path that they want to pass in via `DOCKER_HOST`. 🤔 

I wonder if we can use a similar thing as they did in that update, and switch on the value of `DOCKER_HOST`. And the default can be to try to connect via sockets. I hope this makes sense - it was a bit tricky to unwind, but I'll push a commit up that should hopefully solve this issue.